packages/core/src/transport/WsOutboundTransport.ts

```diff
@@ -101,7 +102,13 @@ export class WsOutboundTransport implements OutboundTransport {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private handleMessageEvent = (event: any) => {
     this.logger.trace('WebSocket message event received.', { url: event.target.url, data: event.data })
-    const payload = JSON.parse(Buffer.from(event.data).toString('utf-8'))
+    const responseMessage = Buffer.from(event.data).toString()
+    const payload = JSON.parse(responseMessage)
```

Comment:
For this we can also use the JsonEncoder:

```suggestion
    const payload = JsonEncoder.fromBuffer(event.data)
```